Quick_Deploy/vLLM/model_repository/vllm/1/model.py

```diff
@@ -0,0 +1,330 @@
+            request = CompletionRequest(**json.loads(serialized_json))
+
+            return (
+                request.prompt,
+                self.make_sample_params(request),
+                self.make_output_options(request),
+            )
+        except Exception as e:
+            self.logger.log_info(f"Error parsing request: {e}")
+            raise e
+
+    def postprocess(
+        self, output: VLLMOutput, output_options: OutputOptions, inflight_avg: float
+    ):
+        """
+        Converts a VLLM output into a Triton response.
+        """
+        # create a Triton Response from the vLLM output
+        response = CompletionResponse()
+        response.request_id = output.request_id
+        response.finished = output.finished
+
+        # number of tokens in the prompt and total number of tokens generated across all completions
+        response.prompt_token_count = len(output.prompt_token_ids)
+
+        # optionally echo the prompt back to the client
+        if output_options.echo:
+            response.prompt = output.prompt
+
+        for completion in output.outputs:
+            completion_output = Completion(
+                index=completion.index,
+                text=completion.text,
+                finish_reason=completion.finish_reason,
+                cumulative_logprob=completion.cumulative_logprob,
+                gen_token_count=len(completion.token_ids),
+            )
+            response.completions.append(completion_output)
+
+        response.current_inflight_count = self._inflight_counter
+        response.average_inflight_count = inflight_avg
+
+        # serialize the NemoResponse into a tensor packed into Triton InferenceResponse
+        serialized_response = response.json().encode("utf-8")
+        triton_output_tensor = pb_utils.Tensor(
+            "serialized_response_json", np.array([serialized_response], dtype=np.object)
+        )
+
+        return pb_utils.InferenceResponse(output_tensors=[triton_output_tensor])
+
+    async def generate_stream(self, request):
+        """
+        Generates a stream of parital outputs from a single request.
+        """
+        try:
+            request_id = request.request_id()
+            prompt, sampling_params, output_options = self.preprocess(request)
+            response_sender = request.get_response_sender()
+            forward_passes = 0
+            inflight_total = 0
+
+            async for output in self.llm.generate(prompt, sampling_params, request_id):
+                forward_passes += 1
+                inflight_total += self._inflight_counter
+                inflight_avg = inflight_total / forward_passes
+                if output_options.stream or output.finished:
+                    response_sender.send(
+                        self.postprocess(output, output_options, inflight_avg)
+                    )
+
+            # TODO: Improve the API for sending FINAL flag; perhaps .complete() or .finalize()
```

Comment:
Question: Is this a note for the response sender api ?

Reply:
Yes